src/comm/protocol.rs

```diff
@@ -138,12 +139,9 @@ impl Connection for UnixStream {
         Box::pin(UnixStream::connect(addr))
     }
 
-    fn addr(&self) -> Self::Addr {
+    fn addr(&self) -> Result<Self::Addr, io::Error> {
         self.peer_addr()
-            .unwrap()
-            .as_pathname()
-            .unwrap()
-            .to_path_buf()
+            .map(|a| a.as_pathname().unwrap().to_path_buf())
```

Comment:
I know this isnt the context where we found the error but should we use expect / check for error instead of this call to unwrap too?

Reply:
I'm a fan of this too. I tried to look up how `as_pathname()` could fail, couldn't figure it out in a few minutes, walked away. But, an `expect("something")` would at least document what is up locally so that readers don't need to wonder.

Reply:
Ack, rewritten with an explanatory comment about why we expect `as_pathname` to return `Some`.